app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalAuthenticateController.php

```diff
@@ -48,11 +51,20 @@ class KnowledgeManagementPortalAuthenticateController extends Controller
     public function __invoke(KnowledgeManagementPortalAuthenticateRequest $request, PortalAuthentication $authentication): JsonResponse
     {
         if ($authentication->isExpired()) {
+            if (! Session::exists('guest_id') && ArticleWasHelpful::active()) {
```

Comment:
I would swap this order to check the Feature Flag first, so that the if statement doesn't have to do any additional operations past that check.

Reply:
Done